### PR TITLE
Add focusin, focusout events to DocumentEventMap/HTMLElementEventMap

### PR DESCRIPTION
**Pull request by lankaapura**
Fixes https://github.com/Microsoft/TypeScript/issues/21822

### CODE REVIEW

File: inputfiles/browser.webidl.xml [hunk 1]
```diff
@@ -3560,6 +3560,8 @@
         <property name="onended" event-handler="ended" type="EventHandler"/>
         <property name="onerror" event-handler="error" type="EventHandler"/>
         <property name="onfocus" event-handler="focus" type="EventHandler"/>
+        <property name="focusin" event-handler="focusin" type="EventHandler"/>
```
Comment (mhegazy) [inline inputfiles/browser.webidl.xml]: these files are autogrenerated and we do not manually change them.So we will need to add support for adding even handler maps to the generation script and add them to overridingTypes.json instead.

Comment (lankaapura) [reply inputfiles/browser.webidl.xml]: @mhegazy thanks, I'll update the PR.

Comment (lankaapura) [reply inputfiles/browser.webidl.xml]: Hi @mhegazy 

I have added following to addedTypes.json but it doesn't generate anything. Do I have to update somewhere else too? I suspect that it's the double quotes in DocumentEventMap properties. (ex: `"blur": FocusEvent;` ) but not sure.

```JavaScript
{
        "kind": "property",
        "interface": "DocumentEventMap",
        "name": "\"focusin\"",
        "type": "FocusEvent"
    },
    {
        "kind": "property",
        "interface": "DocumentEventMap",
        "name": "\"focusout\"",
        "type": "FocusEvent"
    }

```

**Comment by saschanaz** [comment]
Type additions for `*EventMap` interfaces are currently ignored, this is why the test passes without any baseline updates.

**Comment by lankaapura** [comment]
Thanks @saschanaz so do I need to update TS.fsx file to include it? or any other suggestions to fix the issue?

**Comment by saschanaz** [comment]
TS.fsx update is needed.

Sent from my Windows 10 phone

**Comment by lankaapura** [comment]
@saschanaz @mhegazy  PR updated. Pls review.

**Comment by dumistoklus** [comment]
Why this PR not in master branch still?

**Comment by chicoxyzzy** [comment]
They are still not there https://github.com/Microsoft/TSJS-lib-generator/blob/master/src/emitter.ts#L27-L38

**Comment by stevenyxu** [comment]
I also want focusin/focusout because it's inconvenient to keep on writing addEventListener('focusin' as any as keyof HTMLElementEventMap, ...).

But HTMLElementEventMap is generated from https://github.com/Microsoft/TSJS-lib-generator which requires that the event be in a W3C candidate recommendation or further, but I could only find focusin in a working draft https://www.w3.org/TR/uievents/#event-type-focusin. I don't know that it will make it based on the current standard for inclusion.

**Comment by anyexinglu** [comment]
So guys, which version has the 'focusin' and 'focusout'?

**Comment by saschanaz** [comment]
#696 added them again, so I think the next release will have it.

**Pull request merged**